workflows/detaxizer.nf

```diff
@@ -185,19 +189,22 @@ workflow DETAXIZER {
     //
     // MODULE: Prepare Kraken2 Database
     //
-
+    ch_kraken2_db_with_meta = Channel.fromPath(params.kraken2db).map {
+            item -> [['id': "kraken2_db"], item]
+            }
     KRAKEN2PREPARATION (
-        params.kraken2db
+        ch_kraken2_db_with_meta
     )
     ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions)
 
-    // 
+    //
     // MODULE: Run Kraken2
     //
+    ch_kraken2_db = FASTP.out.reads.combine(KRAKEN2PREPARATION.out.db).map{ it -> [it[3]]}
```

Comment:
```suggestion
```

Reply:
I'm actually not quite sure, why do you combine the db with the reads and then use only the db?

Reply:
because otherwise only one process of kraken2 is executed. The combine operator resolves this issue. And only the db is accepted by the kraken2 module. Not quite sure if the nf-core module needs an update so that meta is also used for the db ...

Reply:
Ah okay, so you need a channel of the db which has the same number of items, as the FASTP. This is more Channel magic as I know, but is there a way to have this as a "constant" instead of a channel?

Reply:
in nf-core/mag it seems to be working without that work-around: https://github.com/nf-core/mag/blob/e72890089a1fb69d32f2cf0762739c824412bf2b/workflows/mag.nf#L509-L512 & with a local module https://github.com/nf-core/mag/blob/2.5.1/modules/local/kraken2.nf but might work.

A constant channel is a [value channel](https://www.nextflow.io/docs/latest/channel.html#value-channel), i.e. add `.first()` or such. Which can give an ugly warning.